spec/javascripts/packages/document-capture/components/acuant-capture-canvas-spec.jsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { Provider as AcuantContextProvider } from '@18f/identity-document-capture/context/acuant';
+import AcuantCaptureCanvas from '@18f/identity-document-capture/components/acuant-capture-canvas';
+import render from '../../../support/render';
+import { useAcuant } from '../../../support/acuant';
+
+describe('document-capture/components/acuant-capture-canvas', () => {
+  const { initialize } = useAcuant();
+
+  it('waits for initialization', () => {
+    render(
+      <AcuantContextProvider sdkSrc="about:blank">
+        <AcuantCaptureCanvas />
+      </AcuantContextProvider>,
+    );
+
+    initialize();
+
+    expect(window.AcuantCameraUI.start.calledOnce).to.be.true();
+  });
```

Comment:
this is mostly to check my understanding, but is it worth adding a test that makes sure it's called zero times if `initialize()` is not called?

Reply:
> this is mostly to check my understanding, but is it worth adding a test that makes sure it's called zero times if `initialize()` is not called?

Indirectly, it does test this behavior, based on the fact that in the scenario that it _would_ try to call `initialize`, an error would be thrown, because it's `initialize` which sets those globals in the first place.

```diff
diff --git a/app/javascript/packages/document-capture/components/acuant-capture-canvas.jsx b/app/javascript/packages/document-capture/components/acuant-capture-canvas.jsx
index c73c8f733..a195d9a71 100644
--- a/app/javascript/packages/document-capture/components/acuant-capture-canvas.jsx
+++ b/app/javascript/packages/document-capture/components/acuant-capture-canvas.jsx
@@ -63,17 +63,17 @@ function AcuantCaptureCanvas({
   const { isReady } = useContext(AcuantContext);
 
   useEffect(() => {
-    if (isReady) {
+    // if (isReady) {
       /** @type {AcuantGlobal} */ (window).AcuantCameraUI.start(
         onImageCaptureSuccess,
         onImageCaptureFailure,
       );
-    }
+    // }
 
     return () => {
-      if (isReady) {
+      // if (isReady) {
         /** @type {AcuantGlobal} */ (window).AcuantCameraUI.end();
-      }
+      // }
     };
   }, [isReady]);
```

```
$ npx mocha spec/javascripts/packages/document-capture/components/acuant-capture-canvas-spec.jsx


  document-capture/components/acuant-capture-canvas
    1) waits for initialization
    2) "after each" hook for "waits for initialization"


  0 passing (96ms)
  2 failing

  1) document-capture/components/acuant-capture-canvas
       waits for initialization:
     TypeError: Cannot read property 'start' of undefined
```

I'd actually written this to be more explicit in an earlier iteration, but ran into the exact sort of problem where trying to assert on `window.AcuantCameraUI.start` prior to initialization would throw an error.

```diff
diff --git a/spec/javascripts/packages/document-capture/components/acuant-capture-canvas-spec.jsx b/spec/javascripts/packages/document-capture/components/acuant-capture-canvas-spec.jsx
index 5344f8255..1bf478d6e 100644
--- a/spec/javascripts/packages/document-capture/components/acuant-capture-canvas-spec.jsx
+++ b/spec/javascripts/packages/document-capture/components/acuant-capture-canvas-spec.jsx
@@ -14,6 +14,8 @@ describe('document-capture/components/acuant-capture-canvas', () => {
       </AcuantContextProvider>,
     );
 
+    expect(window.AcuantCameraUI.start.called).to.be.false();
+
     initialize();
 
     expect(window.AcuantCameraUI.start.calledOnce).to.be.true();
```

```
$ npx mocha spec/javascripts/packages/document-capture/components/acuant-capture-canvas-spec.jsx


  document-capture/components/acuant-capture-canvas
    1) waits for initialization
    ✓ ends on unmount


  1 passing (96ms)
  1 failing

  1) document-capture/components/acuant-capture-canvas
       waits for initialization:
     TypeError: Cannot read property 'start' of undefined
```

Maybe a compromise here is just to include an inline code comment which explains the situation above?

Reply:
> Maybe a compromise here is just to include an inline code comment which explains the situation above?

Added in ce7c3babf.